Fail to build Docker image

I want to use OSSGadget with Docker. I follow the instructions in the wiki to build a Docker image in a Ubuntu 18.04 server, but the step 4 failed. The error message is as follows:
```shell
$ docker build -t ossgadget .
Sending build context to Docker daemon  19.59MB
Step 1/5 : FROM mcr.microsoft.com/dotnet/sdk:6.0
6.0: Pulling from dotnet/sdk
bb263680fed1: Pull complete
0ab9fc4a223a: Pull complete
6b08533000e7: Pull complete
af3711477c1c: Pull complete
86426293f679: Pull complete
1618a1d45434: Pull complete
1a7292313abd: Pull complete
8121e72a6dfa: Pull complete
Digest: sha256:85ddcc056581e317e836f372823a60e2af49ac8806130a31b6ce4b9f66ff71c2
Status: Downloaded newer image for mcr.microsoft.com/dotnet/sdk:6.0
 ---> fb2953aaad51
Step 2/5 : COPY . /app/
 ---> 10edc0385c12
Step 3/5 : WORKDIR /app/src
 ---> Running in 9de02620c70d
Removing intermediate container 9de02620c70d
 ---> 7620a61eb863
Step 4/5 : RUN set -o errexit -o nounset     && dotnet build     && ln --symbolic /app/src/oss-characteristics/bin/Debug/net6.0/oss-characteristic /usr/bin/oss-characteristic     && ln --symbolic /app/src/oss-defog/bin/Debug/net6.0/oss-defog /usr/bin/oss-defog     && ln --symbolic /app/src/oss-detect-backdoor/bin/Debug/net6.0/oss-detect-backdoor /usr/bin/oss-detect-backdoor     && ln --symbolic /app/src/oss-detect-cryptography/bin/Debug/net6.0/oss-detect-cryptography /usr/bin/oss-detect-cryptography     && ln --symbolic /app/src/oss-diff/bin/Debug/net6.0/oss-diff /usr/bin/oss-diff     && ln --symbolic /app/src/oss-download/bin/Debug/net6.0/oss-download /usr/bin/oss-download     && ln --symbolic /app/src/oss-find-domain-squats/bin/Debug/net6.0/oss-find-domain-squats /usr/bin/oss-find-domain-squats     && ln --symbolic /app/src/oss-find-source/bin/Debug/net6.0/oss-find-source /usr/bin/oss-find-source     && ln --symbolic /app/src/oss-find-squats/bin/Debug/net6.0/oss-find-squats /usr/bin/oss-find-squats     && ln --symbolic /app/src/oss-health/bin/Debug/net6.0/oss-health /usr/bin/oss-health     && ln --symbolic /app/src/oss-metadata/bin/Debug/net6.0/oss-metadata /usr/bin/oss-metadata     && ln --symbolic /app/src/oss-risk-calculator/bin/Debug/net6.0/oss-risk-calculator /usr/bin/oss-risk-calculator
 ---> Running in 6c278b121923
MSBuild version 17.3.2+561848881 for .NET
  Determining projects to restore...
/app/src/Shared/Shared.Lib.csproj : warning NU1504: Duplicate 'PackageReference' items found. Remove the duplicate items or use the Update functionality to ensure a consistent restore behavior. The duplicate 'PackageReference' items are: Nerdbank.GitVersioning 3.5.119, Nerdbank.GitVersioning 3.4.240.
  Failed to download package 'System.Diagnostics.Debug.4.3.0' from 'https://api.nuget.org/v3-flatcontainer/system.diagnostics.debug/4.3.0/system.diagnostics.debug.4.3.0.nupkg'.
  Resource temporarily unavailable (nuget.cdn.azure.cn:443)
    Resource temporarily unavailable
  Restored /app/src/oss-tests/oss-tests.csproj (in 28.14 sec).
  Restored /app/src/oss-metadata/oss-metadata.csproj (in 54.92 sec).
  Restored /app/src/oss-defog/oss-defog.csproj (in 54.96 sec).
  Restored /app/src/oss-find-source/oss-find-source.csproj (in 54.98 sec).
  Restored /app/src/oss-reproducible/oss-reproducible.csproj (in 54.96 sec).
  Restored /app/src/oss-detect-backdoor/oss-detect-backdoor.csproj (in 55.02 sec).
  Restored /app/src/oss-risk-calculator/oss-risk-calculator.csproj (in 55.04 sec).
  Restored /app/src/oss-characteristics/oss-characteristic.csproj (in 55.06 sec).
  Restored /app/src/oss-detect-cryptography/oss-detect-cryptography.csproj (in 55.06 sec).
  Restored /app/src/Shared/Shared.Lib.csproj (in 55.1 sec).
  Restored /app/src/oss-diff/oss-diff.csproj (in 55.1 sec).
  Restored /app/src/oss-find-domain-squats/oss-find-domain-squats.csproj (in 55.11 sec).
  Restored /app/src/oss-find-squats/oss-find-squats.csproj (in 55.11 sec).
  Restored /app/src/oss-find-squats-lib/oss-find-squats-lib.csproj (in 55.18 sec).
  Restored /app/src/Shared.CLI/Shared.CLI.csproj (in 55.17 sec).
  Restored /app/src/oss-download/oss-download.csproj (in 55.18 sec).
  Restored /app/src/oss-health/oss-health.csproj (in 55.18 sec).
/app/src/Shared/Shared.Lib.csproj : warning NU1504: Duplicate 'PackageReference' items found. Remove the duplicate items or use the Update functionality to ensure a consistent restore behavior. The duplicate 'PackageReference' items are: Nerdbank.GitVersioning 3.5.119, Nerdbank.GitVersioning 3.4.240.
/app/src/Shared/PackageManagers/CondaProjectManager.cs(42,80): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CargoProjectManager.cs(51,80): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NuGetProjectManager.cs(56,80): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(46,81): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NPMProjectManager.cs(52,78): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(65,58): warning CS8604: Possible null reference argument for parameter 's' in 'DateTime DateTime.Parse(string s)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(69,98): warning CS8604: Possible null reference argument for parameter 'uri' in 'ArtifactUri<PyPIArtifactType>.ArtifactUri(PyPIArtifactType type, string uri, DateTime? uploadTime = null)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(74,96): warning CS8604: Possible null reference argument for parameter 'uri' in 'ArtifactUri<PyPIArtifactType>.ArtifactUri(PyPIArtifactType type, string uri, DateTime? uploadTime = null)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(71,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(60,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NuGetProjectManager.cs(81,13): warning CS0162: Unreachable code detected [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NuGetProjectManager.cs(78,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CargoProjectManager.cs(62,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/Model/Repository.cs(138,37): warning CS0618: 'Repository.WatchersCount' is obsolete: 'WatchersCount returns the same data as StargazersCount. You are likely looking to use SubscribersCount. Update your code to use SubscribersCount, as this field will stop containing data in the future' [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CPANProjectManager.cs(88,32): warning CS8600: Converting null literal or possible null value to non-nullable type. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CondaProjectManager.cs(123,31): warning CS8602: Dereference of a possibly null reference. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(89,63): warning CS8604: Possible null reference argument for parameter 'purl' in 'IAsyncEnumerable<ArtifactUri<GolangArtifactType>> GolangProjectManager.GetArtifactDownloadUrisAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CargoProjectManager.cs(88,59): warning CS8604: Possible null reference argument for parameter 'purl' in 'IAsyncEnumerable<ArtifactUri<CargoArtifactType>> CargoProjectManager.GetArtifactDownloadUrisAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(98,105): warning CS8604: Possible null reference argument for parameter 'purl' in 'IAsyncEnumerable<ArtifactUri<MavenArtifactType>> MavenProjectManager.GetArtifactDownloadUrisAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(231,62): warning CS8604: Possible null reference argument for parameter 'purl' in 'Task<string?> GolangProjectManager.GetInfoMetadataAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(210,78): warning CS8604: Possible null reference argument for parameter 'purl' in 'Task<IEnumerable<string>> MavenProjectManager.EnumerateVersionsAsync(PackageURL purl, bool useCache = true, bool includePrerelease = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(241,68): warning CS8604: Possible null reference argument for parameter 'purl' in 'Task<DateTime?> MavenProjectManager.GetPackagePublishDateAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(273,41): warning CS8602: Dereference of a possibly null reference. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(294,50): warning CS8604: Possible null reference argument for parameter 'json' in 'JsonDocument JsonDocument.Parse(string json, JsonDocumentOptions options = default(JsonDocumentOptions))'. [/app/src/Shared/Shared.Lib.csproj]
  Shared.Lib -> /app/src/Shared/bin/Debug/net6.0/OSSGadget.Shared.Lib.dll
/app/src/Shared.CLI/Helpers/PackageDownloader.cs(291,22): warning CS0169: The field 'PackageDownloader.usingTempDir' is never used [/app/src/Shared.CLI/Shared.CLI.csproj]
  Shared.CLI -> /app/src/Shared.CLI/bin/Debug/net6.0/OSSGadget.Shared.Cli.dll
  oss-find-squats-lib -> /app/src/oss-find-squats-lib/bin/Debug/net6.0/oss-find-squats-lib.dll
/app/src/oss-metadata/MetadataTool.cs(55,22): warning CS0414: The field 'MetadataTool._ShowError' is assigned but its value is never used [/app/src/oss-metadata/oss-metadata.csproj]
  oss-download -> /app/src/oss-download/bin/Debug/net6.0/oss-download.dll
  oss-metadata -> /app/src/oss-metadata/bin/Debug/net6.0/oss-metadata.dll
  oss-find-source -> /app/src/oss-find-source/bin/Debug/net6.0/oss-find-source.dll
  oss-diff -> /app/src/oss-diff/bin/Debug/net6.0/oss-diff.dll
  oss-find-squats -> /app/src/oss-find-squats/bin/Debug/net6.0/oss-find-squats.dll
/app/src/oss-defog/DefoggerTool.cs(264,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-defog/oss-defog.csproj]
  oss-defog -> /app/src/oss-defog/bin/Debug/net6.0/oss-defog.dll
/app/src/oss-health/GitHubHealthAlgorithm.cs(91,62): warning CS0618: 'IMiscellaneousClient.GetRateLimits()' is obsolete: 'This client is being deprecated and will be removed in the future. Use RateLimitClient.GetRateLimits instead.' [/app/src/oss-health/oss-health.csproj]
/app/src/oss-health/HealthTool.cs(115,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-health/oss-health.csproj]
/app/src/oss-health/GitHubHealthAlgorithm.cs(212,46): warning CS0618: 'Repository.WatchersCount' is obsolete: 'WatchersCount returns the same data as StargazersCount. You are likely looking to use SubscribersCount. Update your code to use SubscribersCount, as this field will stop containing data in the future' [/app/src/oss-health/oss-health.csproj]
/app/src/oss-characteristics/CharacteristicTool.cs(95,43): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-characteristics/oss-characteristic.csproj]
/app/src/oss-characteristics/CharacteristicTool.cs(249,49): warning CS8602: Dereference of a possibly null reference. [/app/src/oss-characteristics/oss-characteristic.csproj]
  oss-reproducible -> /app/src/oss-reproducible/bin/Debug/net6.0/oss-reproducible.dll
  oss-health -> /app/src/oss-health/bin/Debug/net6.0/oss-health.dll
/app/src/oss-detect-cryptography/DetectCryptographyTool.cs(577,33): warning CS8604: Possible null reference argument for parameter 'Rule' in 'Issue.Issue(Boundary Boundary, Location StartLocation, Location EndLocation, Rule Rule)'. [/app/src/oss-detect-cryptography/oss-detect-cryptography.csproj]
/app/src/oss-detect-cryptography/DetectCryptographyTool.cs(416,46): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-detect-cryptography/oss-detect-cryptography.csproj]
  oss-characteristic -> /app/src/oss-characteristics/bin/Debug/net6.0/oss-characteristic.dll
  oss-find-domain-squats -> /app/src/oss-find-domain-squats/bin/Debug/net6.0/oss-find-domain-squats.dll
  oss-detect-cryptography -> /app/src/oss-detect-cryptography/bin/Debug/net6.0/oss-detect-cryptography.dll
/app/src/oss-detect-backdoor/DetectBackdoorTool.cs(120,65): warning CS8604: Possible null reference argument for parameter 'text' in 'string Output.Blue(string text)'. [/app/src/oss-detect-backdoor/oss-detect-backdoor.csproj]
  oss-detect-backdoor -> /app/src/oss-detect-backdoor/bin/Debug/net6.0/oss-detect-backdoor.dll
/app/src/oss-risk-calculator/RiskCalculatorTool.cs(231,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-risk-calculator/oss-risk-calculator.csproj]
  oss-risk-calculator -> /app/src/oss-risk-calculator/bin/Debug/net6.0/oss-risk-calculator.dll
/app/src/oss-tests/Properties/Resources.resx : error MSB3103: Invalid Resx file. System.IO.DirectoryNotFoundException: Could not find a part of the path '/app/src/oss-tests/testdata/cargo/api_rand_json'. [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.StreamReader..ctor(String path, Encoding encoding) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Build.Tasks.ResourceHandling.MSBuildResXReader.AddLinkedResource(String resxFilename, Boolean pathsRelativeToBasePath, List`1 resources, String name, String value) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Build.Tasks.ResourceHandling.MSBuildResXReader.ParseData(String resxFilename, Boolean pathsRelativeToBasePath, List`1 resources, Dictionary`2 aliases, XElement elem) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Build.Tasks.ResourceHandling.MSBuildResXReader.ReadResources(Stream s, String filename, Boolean pathsRelativeToBasePath) [/app/src/oss-tests/oss-tests.csproj]
Build FAILED.

/app/src/Shared/Shared.Lib.csproj : warning NU1504: Duplicate 'PackageReference' items found. Remove the duplicate items or use the Update functionality to ensure a consistent restore behavior. The duplicate 'PackageReference' items are: Nerdbank.GitVersioning 3.5.119, Nerdbank.GitVersioning 3.4.240.
/app/src/Shared/Shared.Lib.csproj : warning NU1504: Duplicate 'PackageReference' items found. Remove the duplicate items or use the Update functionality to ensure a consistent restore behavior. The duplicate 'PackageReference' items are: Nerdbank.GitVersioning 3.5.119, Nerdbank.GitVersioning 3.4.240.
/app/src/Shared/PackageManagers/CondaProjectManager.cs(42,80): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CargoProjectManager.cs(51,80): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NuGetProjectManager.cs(56,80): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(46,81): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NPMProjectManager.cs(52,78): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(65,58): warning CS8604: Possible null reference argument for parameter 's' in 'DateTime DateTime.Parse(string s)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(69,98): warning CS8604: Possible null reference argument for parameter 'uri' in 'ArtifactUri<PyPIArtifactType>.ArtifactUri(PyPIArtifactType type, string uri, DateTime? uploadTime = null)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(74,96): warning CS8604: Possible null reference argument for parameter 'uri' in 'ArtifactUri<PyPIArtifactType>.ArtifactUri(PyPIArtifactType type, string uri, DateTime? uploadTime = null)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(71,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(60,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NuGetProjectManager.cs(81,13): warning CS0162: Unreachable code detected [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/NuGetProjectManager.cs(78,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CargoProjectManager.cs(62,60): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/Model/Repository.cs(138,37): warning CS0618: 'Repository.WatchersCount' is obsolete: 'WatchersCount returns the same data as StargazersCount. You are likely looking to use SubscribersCount. Update your code to use SubscribersCount, as this field will stop containing data in the future' [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CPANProjectManager.cs(88,32): warning CS8600: Converting null literal or possible null value to non-nullable type. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CondaProjectManager.cs(123,31): warning CS8602: Dereference of a possibly null reference. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(89,63): warning CS8604: Possible null reference argument for parameter 'purl' in 'IAsyncEnumerable<ArtifactUri<GolangArtifactType>> GolangProjectManager.GetArtifactDownloadUrisAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/CargoProjectManager.cs(88,59): warning CS8604: Possible null reference argument for parameter 'purl' in 'IAsyncEnumerable<ArtifactUri<CargoArtifactType>> CargoProjectManager.GetArtifactDownloadUrisAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(98,105): warning CS8604: Possible null reference argument for parameter 'purl' in 'IAsyncEnumerable<ArtifactUri<MavenArtifactType>> MavenProjectManager.GetArtifactDownloadUrisAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(231,62): warning CS8604: Possible null reference argument for parameter 'purl' in 'Task<string?> GolangProjectManager.GetInfoMetadataAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(210,78): warning CS8604: Possible null reference argument for parameter 'purl' in 'Task<IEnumerable<string>> MavenProjectManager.EnumerateVersionsAsync(PackageURL purl, bool useCache = true, bool includePrerelease = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/MavenProjectManager.cs(241,68): warning CS8604: Possible null reference argument for parameter 'purl' in 'Task<DateTime?> MavenProjectManager.GetPackagePublishDateAsync(PackageURL purl, bool useCache = true)'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/GolangProjectManager.cs(273,41): warning CS8602: Dereference of a possibly null reference. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared/PackageManagers/PyPIProjectManager.cs(294,50): warning CS8604: Possible null reference argument for parameter 'json' in 'JsonDocument JsonDocument.Parse(string json, JsonDocumentOptions options = default(JsonDocumentOptions))'. [/app/src/Shared/Shared.Lib.csproj]
/app/src/Shared.CLI/Helpers/PackageDownloader.cs(291,22): warning CS0169: The field 'PackageDownloader.usingTempDir' is never used [/app/src/Shared.CLI/Shared.CLI.csproj]
/app/src/oss-metadata/MetadataTool.cs(55,22): warning CS0414: The field 'MetadataTool._ShowError' is assigned but its value is never used [/app/src/oss-metadata/oss-metadata.csproj]
/app/src/oss-defog/DefoggerTool.cs(264,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-defog/oss-defog.csproj]
/app/src/oss-health/GitHubHealthAlgorithm.cs(91,62): warning CS0618: 'IMiscellaneousClient.GetRateLimits()' is obsolete: 'This client is being deprecated and will be removed in the future. Use RateLimitClient.GetRateLimits instead.' [/app/src/oss-health/oss-health.csproj]
/app/src/oss-health/HealthTool.cs(115,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-health/oss-health.csproj]
/app/src/oss-health/GitHubHealthAlgorithm.cs(212,46): warning CS0618: 'Repository.WatchersCount' is obsolete: 'WatchersCount returns the same data as StargazersCount. You are likely looking to use SubscribersCount. Update your code to use SubscribersCount, as this field will stop containing data in the future' [/app/src/oss-health/oss-health.csproj]
/app/src/oss-characteristics/CharacteristicTool.cs(95,43): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-characteristics/oss-characteristic.csproj]
/app/src/oss-characteristics/CharacteristicTool.cs(249,49): warning CS8602: Dereference of a possibly null reference. [/app/src/oss-characteristics/oss-characteristic.csproj]
/app/src/oss-detect-cryptography/DetectCryptographyTool.cs(577,33): warning CS8604: Possible null reference argument for parameter 'Rule' in 'Issue.Issue(Boundary Boundary, Location StartLocation, Location EndLocation, Rule Rule)'. [/app/src/oss-detect-cryptography/oss-detect-cryptography.csproj]
/app/src/oss-detect-cryptography/DetectCryptographyTool.cs(416,46): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-detect-cryptography/oss-detect-cryptography.csproj]
/app/src/oss-detect-backdoor/DetectBackdoorTool.cs(120,65): warning CS8604: Possible null reference argument for parameter 'text' in 'string Output.Blue(string text)'. [/app/src/oss-detect-backdoor/oss-detect-backdoor.csproj]
/app/src/oss-risk-calculator/RiskCalculatorTool.cs(231,28): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/app/src/oss-risk-calculator/oss-risk-calculator.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103: Invalid Resx file. System.IO.DirectoryNotFoundException: Could not find a part of the path '/app/src/oss-tests/testdata/cargo/api_rand_json'. [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.StreamReader.ValidateArgsAndOpenPath(String path, Encoding encoding, Int32 bufferSize) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at System.IO.StreamReader..ctor(String path, Encoding encoding) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Build.Tasks.ResourceHandling.MSBuildResXReader.AddLinkedResource(String resxFilename, Boolean pathsRelativeToBasePath, List`1 resources, String name, String value) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Build.Tasks.ResourceHandling.MSBuildResXReader.ParseData(String resxFilename, Boolean pathsRelativeToBasePath, List`1 resources, Dictionary`2 aliases, XElement elem) [/app/src/oss-tests/oss-tests.csproj]
/app/src/oss-tests/Properties/Resources.resx : error MSB3103:    at Microsoft.Build.Tasks.ResourceHandling.MSBuildResXReader.ReadResources(Stream s, String filename, Boolean pathsRelativeToBasePath) [/app/src/oss-tests/oss-tests.csproj]    38 Warning(s)
    1 Error(s)

Time Elapsed 00:01:03.65
The command '/bin/sh -c set -o errexit -o nounset     && dotnet build     && ln --symbolic /app/src/oss-characteristics/bin/Debug/net6.0/oss-characteristic /usr/bin/oss-characteristic     && ln --symbolic /app/src/oss-defog/bin/Debug/net6.0/oss-defog /usr/bin/oss-defog     && ln --symbolic /app/src/oss-detect-backdoor/bin/Debug/net6.0/oss-detect-backdoor /usr/bin/oss-detect-backdoor     && ln --symbolic /app/src/oss-detect-cryptography/bin/Debug/net6.0/oss-detect-cryptography /usr/bin/oss-detect-cryptography     && ln --symbolic /app/src/oss-diff/bin/Debug/net6.0/oss-diff /usr/bin/oss-diff     && ln --symbolic /app/src/oss-download/bin/Debug/net6.0/oss-download /usr/bin/oss-download     && ln --symbolic /app/src/oss-find-domain-squats/bin/Debug/net6.0/oss-find-domain-squats /usr/bin/oss-find-domain-squats     && ln --symbolic /app/src/oss-find-source/bin/Debug/net6.0/oss-find-source /usr/bin/oss-find-source     && ln --symbolic /app/src/oss-find-squats/bin/Debug/net6.0/oss-find-squats /usr/bin/oss-find-squats     && ln --symbolic /app/src/oss-health/bin/Debug/net6.0/oss-health /usr/bin/oss-health     && ln --symbolic /app/src/oss-metadata/bin/Debug/net6.0/oss-metadata /usr/bin/oss-metadata     && ln --symbolic /app/src/oss-risk-calculator/bin/Debug/net6.0/oss-risk-calculator /usr/bin/oss-risk-calculator' returned a non-zero code: 1
```

--- a/src/oss-tests/Properties/Resources.resx
+++ b/src/oss-tests/Properties/Resources.resx
@@ -3,7 +3,7 @@
 <root>
     <xsd:schema id="root" xmlns="" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:msdata="urn:schemas-microsoft-com:xml-msdata">
         <xsd:element name="root" msdata:IsDataSet="true">
-            
+
         </xsd:element>
     </xsd:schema>
     <resheader name="resmimetype">
@@ -103,7 +103,7 @@
         <value>..\TestData\Cargo\rand;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
     </data>
     <data name="cargo.rand.json" type="System.Resources.ResXFileRef, System.Windows.Forms">
-        <value>..\testdata\cargo\api_rand_json;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+        <value>..\TestData\Cargo\api_rand_json;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
     </data>
     <data name="maven_ant_1.6_html" type="System.Resources.ResXFileRef, System.Windows.Forms">
         <value>..\TestData\Maven\maven_ant_1.6.html;System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>

